File: server/domain/models/Indicador.py
from mongoengine import EmbeddedDocument, StringField, ListField, ReferenceField, EmbeddedDocumentField
from marshmallow_mongoengine import ModelSchema, fields
from .Amostra import Amostra, AmostraSchema

class Indicador(EmbeddedDocument):

    nome               = StringField(required=True, max_length=200)
    amostras           = ListField(EmbeddedDocumentField("Amostra"))
    indicadores_filhos = ListField(ReferenceField("Indicador"))

    def __repr__(self):
        return '<Indicador(nome={self.nome!r})>'.format(self=self)

class IndicadorSchema(ModelSchema):
    amostras = fields.Nested(AmostraSchema, many=True)
    class Meta:
        model = Indicador

File: client/src/models/indicador.js
import Amostra from "./amostra";

export default class Indicador {
  constructor({ nome, amostras }) {
    amostras = amostras || [];
    
    this.nome = nome;    
    this.amostras = amostras.map((amostra) => new Amostra({ ...amostra }));
  }
}
